shared/src/lib.rs

```diff
@@ -147,22 +147,34 @@ impl IpNetExt for IpNet {
 
 pub fn update_hosts_file(
     interface: &InterfaceName,
-    hosts_path: &Path,
+    opts: &HostsOpts,
     peers: impl IntoIterator<Item = impl AsRef<Peer>>,
 ) -> Result<(), WrappedIoError> {
+    if opts.no_write_hosts {
+        return Ok(());
+    }
     let mut hosts_builder = HostsBuilder::new(format!("innernet {interface}"));
```

Comment:
formatting preferential supernit (separate the early exit):

```suggestion
    }

let mut hosts_builder = HostsBuilder::new(format!("innernet {interface}"));
```